lib/Migrator/APIDiffMigratorPass.cpp

```diff
@@ -605,8 +605,11 @@ struct APIDiffMigratorPass : public ASTMigratorPass, public SourceEntityWalker {
                                           Arg->getStartLoc().getAdvancedLoc(1));
 
           // Replace "x.getY(" with "x.Y =".
-          Editor.replace(ReplaceRange, (llvm::Twine(Walker.Result.str().
-                                                   substr(3)) + " = ").str());
+          auto Replacement = (llvm::Twine(Walker.Result.str()
+                                          .substr(3)) + " = ").str();
+          Replacement[0] = tolower(Replacement[0]);
```

Comment:
Yep! It walks the string and lowercases all consecutive uppercase letters up to the letter before the first lowercase letter.

`URL` -> `url`
`Current` -> `current`
`URLString` -> `urlString`
`XMLHTTPRequest` -> `xmlhttpRequest` (can't win 'em all)